base/linalg/dense.jl

```diff
@@ -71,7 +71,7 @@ vecnorm2{T<:BlasFloat}(x::Union{Array{T},StridedVector{T}}) =
 function triu!(M::AbstractMatrix, k::Integer)
     m, n = size(M)
     if (k > 0 && k > n) || (k < 0 && -k > m)
-        throw(BoundsError())
+        throw(BoundsError("Requested diagonal, $k, out of bounds in matrix of size ($m,$n)"))
```

Comment:
I think `BoundsError` should take an array and index tuple as arguments. You could just use an `ArgumentError`. It's more flexible.